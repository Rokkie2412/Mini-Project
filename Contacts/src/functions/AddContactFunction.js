import { launchCamera,launchImageLibrary } from "react-native-image-picker"

export const allowAddTrueFalse = (firstname,lastname,age,disable,setDisable) =>{
    if((firstname === '' || lastname === '') || age === ''){
        setDisable(true)
        return(disable)
    }else{
        setDisable(false)
        return(disable)
    }
}

export const addContactToAPI = (first,last,age,photo) =>{
    const postContact = fetch('https://simple-contact-crud.herokuapp.com/contact',{
        method:'POST',
        body:JSON.stringify({
            firstName : first,
            lastName : last,
            age : age,
            photo : photo
        }),
        headers:{
            'content-type' : 'application/json',
        }
    }).then((Response)=>Response.json())
    .then(json => console.log(json))

    return postContact
}

export const openCamera = (imageFromCamera,setImageFromCamera) => {
        
        const option={
            mediaType : 'photo',
            quality : 0.6,
        }
        launchCamera(option,(res)=>{
            if(res.didCancel){
                console.log('User Cancel Image Picker');
            }else if(res.errorCode){
                console.log(res.errorMessage);
            }else{
                const data = res.assets[0]
                setImageFromCamera(data.uri)
                console.log(data.uri)
            }
        })
        return imageFromCamera
    }

export const openGallery = (imageFromGallery,setImageFromGallery) =>{
    const option = {
        mediaType : 'photo',
        quality : 0.6
    }
    launchImageLibrary(option,(res)=>{
        if(res.didCancel){
            console.log('User Cancel Image Picker')
        }else if(res.errorCode){
            console.log(res.errorCode)
        }else{
            const data= res.assets[0]
            setImageFromGallery(data.uri)
            console.log(data.uri)
        }
    })
    return imageFromGallery
}