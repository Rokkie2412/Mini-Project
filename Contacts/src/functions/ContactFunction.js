export const GettingData = (Contact,setContact,setTempContact) =>{
    fetch('https://simple-contact-crud.herokuapp.com/contact')
    .then((response)=>response.json())
    .then((json)=>{
        setContact(json.data)
        setTempContact(json.data)
    })
    .catch((error)=>console.log(error))
    return Contact
}

export const FilterSearch = (text,Contact,setTempContact,setText) => {
    if(text){
        const newData = Contact.filter((item)=>{
            const itemData = item.firstName ? item.firstName.toUpperCase() : "".toUpperCase()
            const textData = text.toUpperCase()
            return itemData.indexOf(textData) > -1
        })
        setTempContact(newData)
        setText(text)
    }else{
        setTempContact(Contact)
        setText(text)
    }
}