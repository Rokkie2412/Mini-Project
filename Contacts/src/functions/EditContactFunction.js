export const editContactToAPI = (first,last,age,photo,id) =>{
    const putContact = fetch(`https://simple-contact-crud.herokuapp.com/contact/${id}`,{
        method:'PUT',
        body:JSON.stringify({
            firstName : first,
            lastName : last,
            age : age,
            photo : photo
        }),
        headers:{
            'content-type' : 'application/json',
        }
    }).then((Response)=>Response.json())
    .then(json => console.log(json))

    return putContact
}