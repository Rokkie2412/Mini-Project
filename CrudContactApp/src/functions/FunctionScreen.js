import axios from 'axios'
import * as React from 'react'
import NetInfo from "@react-native-community/netinfo";
import RNRestart from 'react-native-restart';
import { launchCamera,launchImageLibrary } from "react-native-image-picker"
import styles from '../styles/ContactAppStyle'
import {View,Text,KeyboardAvoidingView,FlatList,Pressable,Image} from 'react-native'

export const getData = (setData:Function,Data:Object,FilteredData:Object,setFilteredData:Function):Function =>{
    axios.get('https://simple-contact-crud.herokuapp.com/contact')
    .then(res=>{
    setFilteredData(res.data.data)
    setData(res.data.data)
    })
    return Data
}

export const addData = (first:string,last:string,age:number,foto:string):void =>{
    axios.post('https://simple-contact-crud.herokuapp.com/contact',{
        'firstName' : first,
        'lastName' : last,
        'age' : age,
        'photo' : foto
    },
    {
        headers:{
            'content-type' : 'application/json',
        }
    })
}

export const putData = (first:string,last:string,age:number,foto:string,id:string):void =>{
    axios.put(`https://simple-contact-crud.herokuapp.com/contact/${id}`,{
        'firstName' : first,
        'lastName' : last,
        'age' : age,
        'photo' : foto
    },
    {
        headers:{
            'content-type' : 'application/json',
        }
    })
}

export const FilterSearch = (text:string,Contact:Object,setTempContact:Function,setText:Function):void => {
    if(text){
        const newData = Contact.filter((item)=>{
            const itemData = item.firstName ? item.firstName.toUpperCase() : "".toUpperCase()
            const textData = text.toUpperCase()
            return itemData.indexOf(textData) > -1
        })
        setTempContact(newData)
        setText(text)
    }else{
        setTempContact(Contact)
        setText(text)
    }
}

export const openCamera = (image:string,setImage:Function):Function => {
        
        const option={
            mediaType : 'photo',
            quality : 0.6,
        }
        launchCamera(option,(res)=>{
            if(res.didCancel){
                console.log('User Cancel Image Picker');
            }else if(res.errorCode){
                console.log(res.errorMessage);
            }else{
                const data = res.assets[0]
                setImage(data.uri)
                console.log(data.uri)
            }
        })
        return image
    }

export const openGallery = (image:string,setImage:Function):Function =>{
    const option = {
        mediaType : 'photo',
        quality : 0.6
    }
    launchImageLibrary(option,(res)=>{
        if(res.didCancel){
            console.log('User Cancel Image Picker')
        }else if(res.errorCode){
            console.log(res.errorCode)
        }else{
            const data= res.assets[0]
            setImage(data.uri)
            console.log(data.uri)
        }
    })
    return image
}

export const AddorEditContactLock = (first:string,last:string,age:string):Function =>{
    let correctfirstname:boolean = false
    let correctlastname:boolean = false
    let correntage:boolean = false
    let errorText:string = ''
    let allowAdd:boolean = false

    if(first.length >= 3){
        correctfirstname = true
    }else if(first.length <3){
        correctfirstname = false
    }

    if(last.length >= 3){
        correctlastname = true
    }else if(last.length <3){
        correctlastname = false
    }

    if(last.length >= 3){
        correctlastname = true
    }else if(last.length <3){
        correctlastname = false
    }

    if(age.length !== 0 ){
        correntage = true
    }else if(last.length === 0 ){
        correntage = false
    }

    if(correctfirstname === true && correctlastname === true && correntage === true){
        allowAdd = true
    }else{
        allowAdd = false
    }

   return allowAdd
}

export const emptyAll = (setImage:Function,setLast:Function,setFirst:Function,setAge:Function):void =>{
    setImage('N/A')
    setLast('')
    setFirst('')
    setAge('')
}

export const FuncRepeat = (text:string,setRepeat:Function):void =>{
    if(text.length === 0){
        setRepeat(1000)
    }else{
        setRepeat(100000)
    }
}

export const CheckInternet = (set:Function):void =>{
    NetInfo.fetch('https://simple-contact-crud.herokuapp.com/contact')
    .then(networkState=>{
        if(networkState.isConnected !== true){
            set(true)
        }
    })
}
export const AddPack = (setmodal:Function,age:string,first,last,image,setImage,setLast,setFirst,setAge,setError):void =>{
        setmodal(false)
        const tempPasre = parseInt(age)
        addData(first,last,tempPasre,image)
        setError('')
        emptyAll(setImage,setLast,setFirst,setAge)
    }

export const Restart = ():void =>{
    RNRestart.Restart()
}
       