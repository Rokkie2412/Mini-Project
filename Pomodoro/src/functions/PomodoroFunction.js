import reactNativeBackgroundTimer from 'react-native-background-timer';

export const displayTimer = (secondLeft) =>{
    let hours = Math.floor(secondLeft/60/60)
    let minutes = Math.floor(secondLeft/60%60)
    let seconds = Math.floor(secondLeft%60)

    let displayHours = hours < 10 ? `0${hours}` : hours
    let displayMins = minutes < 10 ? `0${minutes}` : minutes
    let displaySecs = seconds < 10 ? `0${seconds}` : seconds

    return{
      displayHours,
      displayMins,
      displaySecs
    }
}

export const startTimer = (setSecondLeft) =>{
  reactNativeBackgroundTimer.runBackgroundTimer(()=>{
    setSecondLeft(secs =>{
      if(secs > 0){
        return secs-1
      }else if (secs === 0 ) {
        reactNativeBackgroundTimer.stopBackgroundTimer()
      }
    })
  },1000)
}

export const onchecklimit = (value,setValue) =>{
    const parsedqty = Number.parseInt(value)
    if(Number.isNaN(parsedqty))
      setValue(0)
    else if( parsedqty >59)
      setValue(59)
    else
      setValue(parsedqty)
  }