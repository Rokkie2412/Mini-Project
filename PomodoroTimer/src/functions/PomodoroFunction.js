import reactNativeBackgroundTimer from 'react-native-background-timer';
import PushNotification from "react-native-push-notification";
import { Vibration } from 'react-native';
export const displayTimer = (secondLeft) =>{
    let hours = Math.floor(secondLeft/60/60)
    let minutes = Math.floor(secondLeft/60%60)
    let seconds = Math.floor(secondLeft%60)

    let displayHours = hours < 10 ? `0${hours}` : hours
    let displayMins = minutes < 10 ? `0${minutes}` : minutes
    let displaySecs = seconds < 10 ? `0${seconds}` : seconds

    return{
      displayHours,
      displayMins,
      displaySecs
    }
}

export const startTimer = (setSecondLeft) =>{
  reactNativeBackgroundTimer.runBackgroundTimer(()=>{
    setSecondLeft(secs =>{
      if(secs > 0){
        return secs-1
      }else if (secs === 0 ) {
        reactNativeBackgroundTimer.stopBackgroundTimer()
      }
    })
  },1000)
}

export const onchecklimit = (value,setValue) =>{
    const parsedqty = Number.parseInt(value)
    if(Number.isNaN(parsedqty))
      setValue(0)
    else if( parsedqty >59)
      setValue(59)
    else
      setValue(parsedqty)
  }

export const createChannel = () =>{
    PushNotification.createChannel({
      channelId:'test-channel',
      channelName:'Test Channel',
      vibrate:true
    })
  }

export const handleNotification = (judul,pesan) =>{
    PushNotification.localNotification({
      channelId:"test-channel",
      title:judul,
      message:pesan,
      playSound:true,
      id:1,
      vibrate:true,
      vibration:1000
    })
  }
  
export const handleNotificationLongBreak = () =>{
    PushNotification.localNotification({
      channelId:"test-channel",
      title:'Time to long break!',
      message:"it's time to long break! Use it wisely",
      playSound:true,
      id:1,
      vibrate:true,
      vibration:1000,
    })
  }

export const TimerinNotification = (judul,pesan) =>{
    PushNotification.localNotification({
      channelId:"test-channel",
      title:judul,
      message:pesan,
      playSound:false,
      id:2,
      vibrate:false,
    })
  }