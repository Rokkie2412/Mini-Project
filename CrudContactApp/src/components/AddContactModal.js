/**
 * @format
 * @flow
 */

import * as React from 'react'
import {useState} from 'react'
import {View,KeyboardAvoidingView,Pressable,TextInput,Text,Image} from 'react-native'
import Modal from "react-native-modal";
import type {Add} from './component.type'
import {addData,AddorEditContactLock,emptyAll} from '../functions/FunctionScreen'
import Icon from 'react-native-vector-icons/Ionicons'
import styles from '../styles/ModalAddContact.style'
import AwesomeAlert from './AwesomeAlert';

const AddContactModal = ({showmodal,setmodal}:Add):React.Node =>{
    const [first,setFirst] = useState<string>('')
    const [last,setLast] = useState<string>('')
    const [age,setAge] = useState<string>('')
    const [show,setshow] = useState<boolean>(false)
    const [image,setImage] = useState<string>('N/A')
    const [error,setError] = useState<string>('')

    const AddPack = ():void =>{
        setmodal(false)
        const tempPasre = parseInt(age)
        addData(first,last,tempPasre,image)
        setError('')
        emptyAll(setImage,setLast,setFirst,setAge)
    }


    return(
        <KeyboardAvoidingView
        behavior='height'
        pointerEvents='box-none'
        >
            <Modal
            // avoidKeyboard={true}
            transparent={true}
            isVisible={showmodal}
            >
                <View style={styles.mainContainer}>
                    <Pressable 
                    testID='person'
                    onPress={()=>setshow(!show)}
                    style={styles.personIconView}>
                         {image === "N/A" ? <Icon style={styles.personIcon} name="person"/> :
                        <Image style={styles.imagesPerson} source={{uri:`${image}`}}/>
                        }
                    </Pressable>
                    <View style={styles.TextInputSection}>
                        <TextInput
                            testID='first'
                            placeholderTextColor="#eeeeee" 
                            placeholder="Input first name here"
                            style={styles.textinput}
                            value={first}
                            onChangeText={(temp)=>setFirst(temp)}
                        />
                        <TextInput
                            testID='last'
                            placeholderTextColor="#eeeeee" 
                            placeholder="Input last name here"
                            style={styles.textinput}
                            value={last}
                            onChangeText={(temp)=>setLast(temp)}
                        />
                        <TextInput
                            testID='age'
                            placeholderTextColor="#eeeeee" 
                            placeholder="Age"
                            style={styles.textinputAge}
                            value={age}
                            onChangeText={(temp)=>setAge(temp)}
                            keyboardType="numeric"
                        />
                    </View>
                    <Text style={styles.error}>{error}</Text>
                    <View style={styles.ButtonSection}>
                        <Pressable
                        testID='cancel' 
                        onPress={()=>{
                            setmodal(false)
                            emptyAll(setImage,setLast,setFirst,setAge)
                        }}
                        style={styles.Container}>
                            <Text style={styles.Button}>Cancel</Text>
                        </Pressable>
                        <Pressable
                        onPress={()=>{
                            {
                                AddorEditContactLock(first,last,age) === true?
                                AddPack():
                                setError('*Make sure name have 3 character length  and age between 1-99')
                            }
                        }}
                         style={styles.Container}>
                            <Text style={styles.Button}>Add</Text>
                        </Pressable>
                    </View>
                </View>
                <AwesomeAlert show={show} setshow={setshow} image={image} setimage={setImage}/>
            </Modal>
        </KeyboardAvoidingView>
    )
}

export default AddContactModal