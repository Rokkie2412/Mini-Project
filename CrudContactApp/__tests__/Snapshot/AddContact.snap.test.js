import {create,act} from 'react-test-renderer';
import React from 'react';
import AddContactModal from "../../src/components/AddContactModal";
import {AddorEditContactLock} from "../../src/functions/FunctionScreen"

const func = jest.fn()
const tree = create(<AddContactModal setmodal={func}/>)
const mymodule = require('../../src/components/AddContactModal')

jest.mock('react-native-vector-icons/Ionicons', () => 'Mocked Ionicons');
jest.mock('react-native-modal',()=> "Mocked Modal React Native")
jest.mock('@react-native-community/netinfo',()=>'Mocked Netinfo')
jest.mock('react-native-image-picker',()=>{
    return{
        launchCamera:jest.fn(),
        launchImageLibrary:jest.fn()
    }
})

describe('Testing for Modal Edit Contact', () => {
  describe('Snapshot for Edit Contact', () => {
    it('should render correctly',()=>{
        expect(tree).toMatchSnapshot()
    })
  })
  describe('Button get pressed', () => {
    it('should render correctly',()=>{
        const person = tree.root.findByProps({testID:'person'}).props
        act(()=>person.onPress())
        const cancel = tree.root.findByProps({testID:'cancel'}).props
        act(()=>cancel.onPress())
        expect(pack).toBeCalled()
    })
  })
  describe('Textinput onChangeText', () => {
    it('Text input',()=>{
        const first = tree.root.findByProps({testID:'first'}).props
        act(()=>first.onChangeText())
        const last = tree.root.findByProps({testID:'last'}).props
        act(()=>last.onChangeText())
        const age = tree.root.findByProps({testID:'age'}).props
        act(()=>age.onChangeText())
    })
  })
  
})
