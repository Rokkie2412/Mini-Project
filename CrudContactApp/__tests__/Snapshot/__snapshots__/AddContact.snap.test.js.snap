// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing for Modal Edit Contact Snapshot for Edit Contact should render correctly 1`] = `
<View
  onLayout={[Function]}
  pointerEvents="box-none"
>
  <Mocked Modal React Native
    transparent={true}
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#393E46",
          "borderRadius": 30,
          "flex": 1,
          "height": "60%",
          "justifyContent": "center",
          "marginHorizontal": "5%",
          "marginVertical": "20%",
          "position": "absolute",
          "width": "95%",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={Object {}}
        testID="person"
      >
        <Mocked Ionicons
          name="person"
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#00ADB5",
              "borderRadius": 50,
              "color": "#EEEEEE",
              "fontSize": 50,
              "margin": "9%",
              "padding": 15,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "marginHorizontal": "10%",
          }
        }
      >
        <TextInput
          onChangeText={[Function]}
          placeholder="Input first name here"
          placeholderTextColor="#eeeeee"
          style={
            Object {
              "borderBottomWidth": 1,
              "borderColor": "#eeeeee",
              "marginBottom": "7%",
            }
          }
          testID="first"
          value=""
        />
        <TextInput
          onChangeText={[Function]}
          placeholder="Input last name here"
          placeholderTextColor="#eeeeee"
          style={
            Object {
              "borderBottomWidth": 1,
              "borderColor": "#eeeeee",
              "marginBottom": "7%",
            }
          }
          testID="last"
          value=""
        />
        <TextInput
          keyboardType="numeric"
          onChangeText={[Function]}
          placeholder="Age"
          placeholderTextColor="#eeeeee"
          style={
            Object {
              "alignSelf": "center",
              "borderBottomWidth": 1,
              "borderColor": "#eeeeee",
              "marginBottom": "7%",
              "textAlign": "center",
              "width": "20%",
            }
          }
          testID="age"
          value=""
        />
      </View>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#EEEEEE",
            "fontSize": 7,
          }
        }
      >
        
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-evenly",
            "marginTop": "4%",
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "justifyContent": "space-evenly",
            }
          }
          testID="cancel"
        >
          <Text
            style={
              Object {
                "backgroundColor": "#00ADB5",
                "borderRadius": 50,
                "color": "#EEEEEE",
                "fontSize": 20,
                "fontWeight": "bold",
                "maxWidth": "110%",
                "padding": 10,
                "textAlign": "center",
                "width": "120%",
              }
            }
          >
            Cancel
          </Text>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "justifyContent": "space-evenly",
            }
          }
        >
          <Text
            style={
              Object {
                "backgroundColor": "#00ADB5",
                "borderRadius": 50,
                "color": "#EEEEEE",
                "fontSize": 20,
                "fontWeight": "bold",
                "maxWidth": "110%",
                "padding": 10,
                "textAlign": "center",
                "width": "120%",
              }
            }
          >
            Add
          </Text>
        </View>
      </View>
    </View>
    <Mocked Modal React Native
      isVisible={false}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#EEEEEE",
          "borderRadius": 30,
          "flex": 1,
          "height": "10%",
          "marginVertical": "50%",
          "position": "absolute",
          "width": "60%",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID="Close"
      >
        <Mocked Ionicons
          name="close"
          style={
            Object {
              "alignSelf": "flex-end",
              "backgroundColor": "#222831",
              "borderRadius": 50,
              "color": "#EEEEEE",
              "marginBottom": 20,
              "marginRight": 10,
              "padding": 8,
            }
          }
        />
      </View>
      <Text
        style={
          Object {
            "color": "#222831",
            "fontSize": 18,
            "fontWeight": "600",
            "textAlign": "center",
          }
        }
      >
        Choose image source :
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-evenly",
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="openCamera"
        >
          <Mocked Ionicons
            name="camera"
            style={
              Object {
                "backgroundColor": "#222831",
                "borderRadius": 50,
                "color": "#EEEEEE",
                "fontSize": 24,
                "margin": 10,
                "padding": "10%",
              }
            }
          />
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="openGallery"
        >
          <Mocked Ionicons
            name="image"
            style={
              Object {
                "backgroundColor": "#222831",
                "borderRadius": 50,
                "color": "#EEEEEE",
                "fontSize": 24,
                "margin": 10,
                "padding": "10%",
              }
            }
          />
        </View>
      </View>
    </Mocked Modal React Native>
  </Mocked Modal React Native>
</View>
`;
